tardis/transport/montecarlo/packet_trackers.py

```diff
@@ -194,15 +251,33 @@ def __init__(self):
         self.interaction_type = -1
 
     def track(self, r_packet):
+        """
+        Track properties of RPacket and override the previous values
+        """
         self.index = r_packet.index
         self.r = r_packet.r
         self.nu = r_packet.nu
         self.energy = r_packet.energy
         self.shell_id = r_packet.current_shell_id
         self.interaction_type = r_packet.last_interaction_type
 
-    # To make it compatible with RPacketTracker
     def finalize_array(self):
+        """
+        Added to make RPacketLastInteractionTracker compatible with RPacketTracker
+        """
+        pass
+
+    # To make it compatible with RPacketTracker
+    def track_boundary_interaction(self, current_shell_id, next_shell_id):
```

Comment:
This

Reply:
But it is a function of different class.

Reply:
> This

In that case, we also had two `finalize_array`, `track` functions before but the benchmark passed.

Reply:
> But it is a function of different class.

oh sorry didn't notice that